test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -779,6 +808,38 @@ protected static void wipeAllIndices() throws IOException {
         }
     }
 
+    // The watcher index may be recreated between when we reset system features and when
+    // we call a wildcard deletion
```

Comment:
Can we expand this comment with why we think this happens, and that this can be removed once `DELETE *` doesn't hit system indices anyway?